### PR TITLE
Turn on auth for ironic json-rpc

### PR DESCRIPTION
**Pull request by honza**
The companion to https://github.com/openshift/machine-api-operator/pull/727

### CODE REVIEW

**Comment by hardys** [comment]
Do we need to go ahead with https://github.com/openshift/machine-api-operator/pull/727 now that CBO is enabled by default, or just make this change?

**Comment by asalkeld** [comment]
this looks good to me..

**Review by dtantsur — APPROVED** [review]

File: provisioning/baremetal_pod.go [hunk 1]
```diff
@@ -218,6 +238,7 @@ func createContainerMetal3BaremetalOperator(images *Images, config *metal3iov1al
 		ImagePullPolicy: "IfNotPresent",
 		VolumeMounts: []corev1.VolumeMount{
 			ironicCredentialsMount,
+			rpcCredentialsMount,
```
Comment (zaneb) [inline provisioning/baremetal_pod.go:241]: baremetal-operator doesn't need RPC creds. It's an interface that is purely internal to Ironic.

**Comment by sadasu** [comment]
Can we add any unit tests here?

File: provisioning/baremetal_pod.go (hunk 1 above)
Comment (honza) [reply provisioning/baremetal_pod.go:241]: Derek mentioned in the original PR that Ironic needs the creds to talk to itself?

Comment (dtantsur) [reply provisioning/baremetal_pod.go:241]: Ironic Conductor, but not BMO.

Comment (honza) [reply provisioning/baremetal_pod.go]: Done

**Comment by sadasu** [comment]
I am ready to approve. Could you please add a unit test to baremetal_secrets_test.go here https://github.com/openshift/cluster-baremetal-operator/blob/master/provisioning/baremetal_secrets_test.go#L65, like we do for the other secrets?

**Comment by honza** [comment]
> Could you please add a unit test to baremetal_secrets_test.go here https://github.com/openshift/cluster-baremetal-operator/blob/master/provisioning/baremetal_secrets_test.go#L65, like we do for the other secrets?

Done

**Comment by zaneb** [comment]
It's so weird to me that every repo managed by Prow has different rules for e.g. whether /lgtm from an approver counts as approval. Anyway...
/approve

**Comment by sadasu** [comment]
/override ci/prow/e2e-agnostic

the same 5 e2e-agnostic tests failing. So, OK to override.

**Pull request merged**